main.go

```diff
@@ -105,14 +103,18 @@ func main() {
 
 	restConfig := client.GetConfigOrDie(clientOptions)
 	mgr, err := ctrl.NewManager(restConfig, ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		HealthProbeBindAddress: healthAddr,
-		Port:                   9443,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "5b6ca942.fluxcd.io",
-		Namespace:              watchNamespace,
-		Logger:                 ctrl.Log,
+		Scheme:                        scheme,
+		MetricsBindAddress:            metricsAddr,
+		HealthProbeBindAddress:        healthAddr,
+		Port:                          9443,
+		LeaderElection:                leaderElectionOptions.Enable,
+		LeaderElectionReleaseOnCancel: leaderElectionOptions.ReleaseOnCancel,
+		LeaseDuration:                 &leaderElectionOptions.LeaseDuration,
+		RenewDeadline:                 &leaderElectionOptions.RenewDeadline,
+		RetryPeriod:                   &leaderElectionOptions.RetryPeriod,
+		LeaderElectionID:              "5b6ca942.fluxcd.io",
```

Comment:
Something that comes to mind now that we are touching leader election configurations is to change the ID to something more identifying, as the current IDs have been autogenerated by kubebuilder.

[CAPI seems to use lengthy strings](https://github.com/kubernetes-sigs/cluster-api/search?q=LeaderElectionID), but I think `helm-controller.fluxcd.io` would also suffice.

Reply:
Other option would be `flux-helm-controller-leader-election`.